### DISCUSSION TITLE
Accessing DB directly with NTS - Multiple DBs?

### DISCUSSION THREAD

**Discussion by Taalha** [NetTopologySuite/NetTopologySuite#680] [Q&A]
In an attempt to replace what Esris SDK does I found this library. One thing is not clear to me however. Which DBs or data sources in general (like Shape files or Geopackages) does NTS support?

If I want to support PostGIS as well as MySQL in my application, can I use NTS to retrieve data from them and work with it?

Or do I have to load in my data and then work with my data using NTS? This seems bad as that would potentially mean having to query tons of data.

**Comment by DGuidi** [comment]
please take a look ath the *IO* projects that deal with data access https://github.com/NetTopologySuite

> Or do I have to load in my data and then work with my data using NTS? This seems bad as that would potentially mean having to query tons of data

unless you use PostGIS functions tools (or similar database-integrated librarties), AFAIK this is the same potential issue that you have with ESRI SDK. 
maybe you can take a look at [EF.Spatial](https://learn.microsoft.com/en-us/ef/core/modeling/spatial) too

**Reply by FObermaier** [reply]
> maybe you can take a look at [EF.Spatial](https://learn.microsoft.com/en-us/ef/core/modeling/spatial) too.

Or NHibernate.Spatial
